models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -66,42 +69,60 @@ WITH
                 , call_block_time
                 , call_block_number
                 , call_success
-                , nftRecipient as call_from
-                , 'Buy' as trade_category 
+                , nftRecipient as trade_recipient
+                , 'Buy' as trade_category
+                , 'isRouter' as called_from_router
             FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForSpecificNFTs') }}
             WHERE call_success = true
             {% if is_incremental() %}
             AND call_block_time >= (select max(block_time) from {{ this }})
             {% endif %}
         ) s
-    ),
+    )
+
+    -- this join should be removed in the future when more call trace info is added to the _call_ tables, we need the call_from field to track down the eth traces.
+    , swaps_with_calldata as (
```

Comment:
why are we pulling this data here instead of later on in swaps_with_traces? seems expensive to join twice on ethereum.traces

Reply:
We are joining on different conditions here so I don't think there's a way around it.
Here we need the tracedata of the trace that corresponds to the call, later we need all the traces from the subtraces that came from the call. 
We should push for more standard trace data to be included into the function call tables by default (like `from` and `to` fields). That would make this join not needed anymore (as stated in the inline comment)

Reply:
ah I see, didn't notice that you changed `call_from` to `trade_recipient` in the earlier CTE. makes sense, yeah would be nice to have `call_from`, `call_to`, and `value` in the decoded tables.

Reply:
yeah that field name was causing me lots of confusions. :laughing: